### PR TITLE
430523 token check

### PR DESCRIPTION
**Pull request by whitewaterdesign**

### CODE REVIEW

**Comment by whitewaterdesign** [comment]
Just a heads up, there's a contract test update on portal - this won't succeed until portal is merged.

File: app/plugins/auth.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        ],
+        verify: {
+          aud: 'aphw-ddi-api',
+          iss: [issuers.portal, issuers.enforcement],
+          sub: false
+        },
+        validate
+      })
+      server.auth.default('api_jwt')
+
```
Comment (jbarnsley10) [inline app/plugins/auth.js:33]: This can be removed

File: app/plugins/cron.js [hunk 2]
```diff
@@ -11,6 +12,15 @@ const setupCron = (server) => {
     scheduled: true,
     timezone: 'Europe/London'
   })
+
+  cron.schedule(config.purgeExpiredCacheCrontab ?? '0 * * * *', async () => {
```
Comment (jbarnsley10) [inline app/plugins/cron.js:16]: I'd suggest every 2 or more hours

File: app/repos/people.js [hunk 3]
```diff
@@ -555,6 +555,9 @@ const purgePersonByReferenceNumber = async (reference, user, transaction) => {
 
   await person.destroy({ force: true, transaction })
```
Comment (jbarnsley10) [inline app/repos/people.js:557]: Needs moving to before the person destory

File: app/repos/search-index.js [hunk 4]
```diff
@@ -125,6 +125,7 @@ const removeDogFromSearchIndex = async (dogFromDb, transaction) => {
 
     await indexRow.destroy({ transaction })
   }
+  await sequelize.models.search_tgram.destroy({ where: { dog_id: dogFromDb.id } }, { transaction })
```
Comment (jbarnsley10) [inline app/repos/search-index.js:128]: Needs moving to before search_index destroy

**Review by jbarnsley10 — APPROVED** [review]

**Pull request merged**